### PR TITLE
🌐 Add Chinese translation for Tutorial - Middleware

### PR DESCRIPTION
**Pull request by lpdswing**

### CODE REVIEW

**Comment by lpdswing** [comment]
@tiangolo  I don't know why occured this error when build doc .
image

**Comment by tiangolo** [comment]
Hey @lpdswing ! Thanks for letting me know.

The error had nothing to do with you, it was caused by a recent change I made when I enabled the new features of [Material for MkDocs Insiders](https://squidfunk.github.io/mkdocs-material/insiders/), but I incorrectly handled the install in PRs from forks.

I just fixed it and updated your branch, so it's now passing and ready for review :tada:

**Comment by lpdswing** [comment]
OK,thx发自我的华为手机-------- 原始邮件 --------发件人： Sebastián Ramírez <notifications@github.com>日期： 2020年11月11日周三 凌晨3:56收件人： tiangolo/fastapi <fastapi@noreply.github.com>抄送： lpdswing <lpdswing@qq.com>, Mention <mention@noreply.github.com>主    题： Re: [tiangolo/fastapi] Add chinese translate for tutorial-middleware (#2334)

Hey @lpdswing ! Thanks for letting me know.

The error had nothing to do with you, it was caused by a recent change I made when I enabled the new features of Material for MkDocs Insiders, but I incorrectly handled the install in PRs from forks.

I just fixed it and updated your branch, so it's now passing and ready for review 🎉

File: docs/zh/docs/tutorial/middleware.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
+# 中间件
+
+你可以向 **FastAPI** 应用添加中间件.
+
+"middleware"是一个函数,它在处理每个特定*路径操作*的**request**前工作,并且在每个**response**返回之前工作.
```
Comment (waynerv) [inline docs/zh/docs/tutorial/middleware.md:5]: 「中间件」是一个函数，它在每个**请求**被特定的*路径操作*处理之前以及每个**响应**被返回之前工作.

File: docs/zh/docs/tutorial/middleware.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+* 它接收你的应用程序的每一个**request**.
```
Comment (waynerv) [inline docs/zh/docs/tutorial/middleware.md:7]: `request` and `response` should be translated.

File: docs/zh/docs/tutorial/middleware.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+* 然后它可以对这个**request**做一些事情或者执行任何需要的代码.
+* 然后它将**request**传递给应用程序的其他部分 (通过某种*路径操作*).
```
Comment (waynerv) [inline docs/zh/docs/tutorial/middleware.md:9]: 然后它将**请求**传递给应用程序的其他部分（某个*路径操作*）进行处理。

File: docs/zh/docs/tutorial/middleware.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+* 然后它获取应用程序生产的**response** (通过某种*路径操作*).
```
Comment (waynerv) [inline docs/zh/docs/tutorial/middleware.md:10]: 然后它获取应用程序（某个*路径操作*）生成的**响应** 。

File: docs/zh/docs/tutorial/middleware.md [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+* 它可以对该**response**做些什么或者执行任何需要的代码.
+* 然后它返回这个 **response**.
+
+!!! note "技术细节"
+    如果你依赖了 `yield`, 退出代码将在执行中间件*后*执行.
```
Comment (waynerv) [inline docs/zh/docs/tutorial/middleware.md:15]: 如果你有使用了 `yield` 关键字的依赖，依赖中的退出代码将在执行中间件*之后*执行。

File: docs/zh/docs/tutorial/middleware.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+!!! tip
+    请记住可以 <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers" class="external-link" target="_blank">用'X-' 前缀</a>添加专有自定义请求头.
+
+    但是如果你想让浏览器中的客户端看到你的自定义请求头, 你需要把它们加到 CORS 配置 ([CORS (Cross-Origin Resource Sharing)](cors.md){.internal-link target=_blank}) 的 `expose_headers` 参数中,在 <a href="https://www.starlette.io/middleware/#corsmiddleware" class="external-link" target="_blank">Starlette's CORS docs</a>文档中.
+
+!!! note "技术细节"
+    你也可以使用 `from starlette.requests import Request`.
+
+    **FastAPI** 为开发者提供了方便. 但是你也可以直接使用Starlette.
```
Comment (waynerv) [inline docs/zh/docs/tutorial/middleware.md:43]: **FastAPI** 出于方便开发者的原因提供了该对象，但其实它直接来自于 Starlette。

File: docs/zh/docs/tutorial/middleware.md [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+例如你可以添加自定义请求头 `X-Process-Time` 包含以秒为单位的接收请求和生成相应的时间:
+
+```Python hl_lines="10  12-13"
+{!../../../docs_src/middleware/tutorial001.py!}
+```
+
+## 其他中间件
+
+你可以稍后再 [Advanced User Guide: Advanced Middleware](../advanced/middleware.md){.internal-link target=_blank}阅读更过关于中间件的教程.
```
Comment (waynerv) [inline docs/zh/docs/tutorial/middleware.md:59]: 你可以稍后在 [Advanced User Guide: Advanced Middleware](../advanced/middleware.md){.internal-link target=_blank}阅读更多关于中间件的教程。

**Review by waynerv — CHANGES_REQUESTED** [review]

File: docs/zh/docs/tutorial/middleware.md [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+!!! note "技术细节"
+    如果你使用了 `yield`关键字依赖, 依赖中的退出代码将在执行中间件*后*执行.
+
+    如果有任何后台任务(稍后记录), 它们将在执行中间件*后*运行.
+
+## 创建中间件
+
+要创建中间件你可以在函数的顶部使用装饰器 `@app.middleware("http")`.
+
+中间件参数接收:
```
Comment (RunningIkkyu) [inline docs/zh/docs/tutorial/middleware.md:23]: 中间件函数接收如下参数：

File: docs/zh/docs/tutorial/middleware.md [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+
+* 它接收你的应用程序的每一个**请求**.
+* 然后它可以对这个**请求**做一些事情或者执行任何需要的代码.
+* 然后它将**请求**传递给应用程序的其他部分 (通过某种*路径操作*).
+* 然后它获取应用程序生产的**响应** (通过某种*路径操作*).
+* 它可以对该**响应**做些什么或者执行任何需要的代码.
+* 然后它返回这个 **响应**.
+
+!!! note "技术细节"
+    如果你使用了 `yield`关键字依赖, 依赖中的退出代码将在执行中间件*后*执行.
```
Comment (RunningIkkyu) [inline docs/zh/docs/tutorial/middleware.md:15]: Leave a space after `yield`.

File: docs/zh/docs/tutorial/middleware.md [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+
+## 创建中间件
+
+要创建中间件你可以在函数的顶部使用装饰器 `@app.middleware("http")`.
+
+中间件参数接收:
+
+* `request`.
+* 一个函数 `call_next` 它将接收 `request` 作为参数.
+    * 这个函数将 `request` 传递个相应的 *路径操作*.
```
Comment (RunningIkkyu) [inline docs/zh/docs/tutorial/middleware.md:27]: 传递给

File: docs/zh/docs/tutorial/middleware.md [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+
+!!! tip
+    请记住可以 <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers" class="external-link" target="_blank">用'X-' 前缀</a>添加专有自定义请求头.
+
+    但是如果你想让浏览器中的客户端看到你的自定义请求头, 你需要把它们加到 CORS 配置 ([CORS (Cross-Origin Resource Sharing)](cors.md){.internal-link target=_blank}) 的 `expose_headers` 参数中,在 <a href="https://www.starlette.io/middleware/#corsmiddleware" class="external-link" target="_blank">Starlette's CORS docs</a>文档中.
+
+!!! note "技术细节"
+    你也可以使用 `from starlette.requests import Request`.
+
+    **FastAPI** 为了开发者方便提供了该对象. 但其实它直接来自于Starlette.
```
Comment (RunningIkkyu) [inline docs/zh/docs/tutorial/middleware.md:43]: Leave a space before `Starlette`.

File: docs/zh/docs/tutorial/middleware.md [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+* 然后你可以在返回 `response` 前进一步修改它.
+
+```Python hl_lines="8-9  11  14"
+{!../../../docs_src/middleware/tutorial001.py!}
+```
+
+!!! tip
+    请记住可以 <a href="https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers" class="external-link" target="_blank">用'X-' 前缀</a>添加专有自定义请求头.
+
+    但是如果你想让浏览器中的客户端看到你的自定义请求头, 你需要把它们加到 CORS 配置 ([CORS (Cross-Origin Resource Sharing)](cors.md){.internal-link target=_blank}) 的 `expose_headers` 参数中,在 <a href="https://www.starlette.io/middleware/#corsmiddleware" class="external-link" target="_blank">Starlette's CORS docs</a>文档中.
```
Comment (RunningIkkyu) [inline docs/zh/docs/tutorial/middleware.md:38]: Leave spaces for `Starlette's CORS docs`.

File: docs/zh/docs/tutorial/middleware.md [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+
+!!! note "技术细节"
+    你也可以使用 `from starlette.requests import Request`.
+
+    **FastAPI** 为了开发者方便提供了该对象. 但其实它直接来自于Starlette.
+
+### 在 `response` 的前和后
+
+在任何*路径操作*收到`request`前,可以添加要和请求一起运行的代码.
```
Comment (RunningIkkyu) [inline docs/zh/docs/tutorial/middleware.md:47]: Wrap enlish words with a space.

**Review by RunningIkkyu — CHANGES_REQUESTED** [review]

File: docs/zh/docs/tutorial/middleware.md (hunk 9 above)
Comment (lpdswing) [reply docs/zh/docs/tutorial/middleware.md:15]: 删除空格?

File: docs/zh/docs/tutorial/middleware.md (hunk 10 above)
Comment (lpdswing) [reply docs/zh/docs/tutorial/middleware.md:27]: thx

File: docs/zh/docs/tutorial/middleware.md (hunk 9 above)
Comment (RunningIkkyu) [reply docs/zh/docs/tutorial/middleware.md]: `yield` 后面加个空格

**Review by RunningIkkyu — APPROVED** [review]

File: docs/zh/docs/tutorial/middleware.md [hunk 14]
```diff
@@ -0,0 +1,61 @@
...
+
+例如你可以添加自定义请求头 `X-Process-Time` 包含以秒为单位的接收请求和生成相应的时间:
+
+```Python hl_lines="10  12-13"
+{!../../../docs_src/middleware/tutorial001.py!}
+```
+
+## 其他中间件
+
+你可以稍后再 [Advanced User Guide: Advanced Middleware](../advanced/middleware.md){.internal-link target=_blank}阅读更多关于中间件的教程.
```
Comment (waynerv) [inline docs/zh/docs/tutorial/middleware.md:59]: 再 -> 在

File: docs/zh/docs/tutorial/middleware.md [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+
+    **FastAPI** 为了开发者方便提供了该对象. 但其实它直接来自于 Starlette.
+
+### 在 `response` 的前和后
+
+在任何*路径操作*收到`request`前,可以添加要和请求一起运行的代码.
+
+也可以在*响应*生成但是返回之前添加代码.
+
+例如你可以添加自定义请求头 `X-Process-Time` 包含以秒为单位的接收请求和生成相应的时间:
```
Comment (waynerv) [inline docs/zh/docs/tutorial/middleware.md:51]: 相应 -> 响应

**Review by waynerv — CHANGES_REQUESTED** [review]
You should also translate all the punctuation marks.

**Review by maoyibo — APPROVED** [review]

**Comment by tiangolo** [comment]
Awesome! Thank you @lpdswing for your contribution! :cake: :rocket: 

And thanks @waynerv, @RunningIkkyu, and @maoyibo for the reviews :bow: :coffee: 

---

I'm going to wait for the punctuation fixes and then the approval by @waynerv before merging :nerd_face: :memo:

**Review by waynerv — APPROVED** [review]
@tiangolo I think this PR is ready for merging.

**Comment by tiangolo** [comment]
Excellent, thanks for the confirmation @waynerv ! :rocket:

**Pull request merged**